bundles/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/HPPrinterHandler.java

```diff
@@ -100,12 +102,16 @@ protected void updateStatus(final ThingStatus status) {
 
     @Override
     public void channelLinked(ChannelUID channelUID) {
-        binder.channelsChanged();
+        final HPPrinterBinder localBinder = binder;
+        if (localBinder != null)
+            localBinder.channelsChanged();
```

Comment:
It's good to prevent NPEs this way. Though it's also a best practice to prevent bugs by always adding the curly braces for single line if statements. Probably your IDE can insert them automatically on saves? Please also check the other occurrences.

```suggestion
        if (localBinder != null) {
            localBinder.channelsChanged();
        }
```

Reply:
I'll have to look into the VSCode/Java addon settings for this, regardless I've gone through all the code and made sure single line if statements have curly braces.